lemminx: Add Java crypto module for ECC HTTPS

tricktron opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Add the Java crypto module so SSL/HTTPS connections using elliptic curves succeed. Without elliptic-curve support, those connections fail during download.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Aleksanaa commented:
I don't understand what is special about lemminx here. Should we add this to all java applications?

tricktron commented:
@Aleksanaa There is nothing special about lemminx. I just have not thought about adding this for all Maven packages. The only required precondition is that the `jvmArgs` must be passed **before** the `-jar <your jar>` argument.

Currently, most Java apps use the `maven.buildMavenPackage` function and override the installPhase and wrap the jar with `makeWrapper`. If you see an easy way to generalize this for all Java apps, that would be even better. I just had this simple change laying around in my nixos config and wanted to upstream it.

tricktron committed:
lemminx: Add Java crypto module for ECC HTTPS

Add the Java crypto module so SSL/HTTPS connections using elliptic curves
succeed. Without elliptic-curve support, those connections fail during
download.

tricktron commented:
> Does JAVA_TOOL_OPTIONS work for you?
> 
> I still feel it quite absurd to set flags at build time just to get a specific proxy

@Aleksanaa  Yes that works like a charm, thank you. I then found the underlying problem: Our corporate proxy was using elliptic curves during the ssl handshake and the corresponding java module was missing in the used minimal jre. So I adapted this pr and added the ec crypto module to fix it. What do you think?
